src/pages/components/MapLayers/BusToolTip.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react'
+import { Point } from 'src/pages/RealtimeMapPage'
+import { Button, Grid } from '@mui/material'
+import { DivIcon } from 'leaflet'
+import moment from 'moment-timezone'
+
+import { getSiriRideWithRelated } from 'src/api/siriService'
+import { SiriRideWithRelatedPydanticModel } from 'open-bus-stride-client/openapi/models/SiriRideWithRelatedPydanticModel'
+import { Row } from 'src/pages/components/Row'
+import { Label } from 'src/pages/components/Label'
+import { TEXTS } from 'src/resources/texts'
+import { Spin } from 'antd'
+
+export function BusToolTip({ position, icon }: { position: Point; icon: DivIcon }) {
+  const [siriRide, setSiriRide] = useState<SiriRideWithRelatedPydanticModel | undefined>()
+  const [isLoading, setIsLoading] = useState(false)
+  const [showJson, setShowJson] = useState(false)
+
+  useEffect(() => {
+    setIsLoading(true)
+    getSiriRideWithRelated(
+      position.point!.siri_route__id.toString(),
+      position.point!.siri_ride__vehicle_ref.toString(),
+      position.point!.siri_route__line_ref.toString(),
+    )
+      .then((siriRideRes: SiriRideWithRelatedPydanticModel) => setSiriRide(siriRideRes))
+      .finally(() => setIsLoading(false))
+  }, [position])
+  return (
+    <div style={showJson ? { minWidth: '30rem', background: 'white' } : {}}>
+      {isLoading ? (
+        <Row>
+          <Label text={TEXTS.loading_routes} />
+          <Spin />
+        </Row>
+      ) : showJson ? (
+        <pre>
+          {JSON.stringify(position, null, 2)}
+          <br />
+          {siriRide && JSON.stringify(siriRide, null, 2)}
+        </pre>
+      ) : (
+        <Grid container display="flex" direction="row-reverse" justifyContent="space-between">
+          <Grid item xs={4}>
```

Comment:
let's avoid abusing the grid system - it's not necessary here. we can apply CSS. it can be on seperate PR